### DISCUSSION TITLE
Formattare l'orario

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#345] [HUB: Home Assistant]
Ora e minuti non mi restituisce nulla. Mi suggerite dove sta l'errore?
```
- platform: template
    sensors:
      data_formattata:
        friendly_name: "Data italiana"
        value_template: >
          {% set numgiorno = ["Lunedì", "Martedì", "Mercoledì", "Giovedì", "Venerdì", "Sabato", "Domenica"] %}
          {% set giornset = numgiorno[now().weekday()] %}
          {% set nummese = ["01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12"] %}
          {% set tempo = states("sensor.time") %}
          {% set ora = as_timestamp(tempo) | timestamp_custom('%H') | int %}
          {% set minuti = as_timestamp(tempo) | timestamp_custom('%M') | int %}
          {% set mese = nummese[now().month] %}
          {{ giornset }} {{ now().day }}/{{ mese }}/{{ now().year }} {{ ora }}:{{ minuti }}
```

**Comment by Larry-78** [comment]
Ciao, forse non funziona perché stai cercando di ottenerli attraverso la variabile tempo che hai impostato precedentemente e non so se quella forma è corretta.
Ore e minuti potresti ottenerli semplicemente in questo modo
`{{now().strftime("%H")}}`
`{{now().strftime("%M")}}`
... oppure per ottenere una forma più completa con giorno, mese, anno e orario con i numeri, potresti utilizzare questa
`{{now().strftime("%d/%m/%Y - %H:%M:%S")}}`

**Reply by xyzmauro** [reply]
l'unica cosa che viene dal di fuori del template è sensor.time, come lo hai definito?

**Reply by Dregi56** [reply]
```
- platform: time_date
    display_options:
      - 'date_time'
      - 'time'
```
Comunque col suggerimento di larry-78 funziona bene. Grazie
